internal/services/mysql/mysql_flexible_server_resource_test.go

```diff
@@ -568,6 +568,35 @@ func TestAccMySqlFlexibleServer_updateToWriteOnlyPassword(t *testing.T) {
 	})
 }
 
+func TestAccMySqlFlexibleServer_updatePublicNetworkAccess(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mysql_flexible_server", "test")
+	r := MySqlFlexibleServerResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.publicNetworkAccess(data, "null"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
```

Comment:
Are these `Check`s made unnecessary by the `ImportStep`s?

Reply:
We have it here to confirm it [exists](https://github.com/hashicorp/terraform-provider-azurerm/pull/28890/files#diff-d24863cabe3a5a3e3e6acadc357ca64520901f1ffc25ce0bf48104065590a8b0R600) in Azure. Sometimes it doesn't even though it created because Azure is weird. You're probably right that we don't need it if we're checking import but it only costs an extra api call to doubly make sure it at least exists in Azure before we move on. Similar to how we had those old checks that we don't do anymore, we might move in that direction of removing this as well but for now, let's keep it consistent with how we do other tests

Reply:
Roger. I was considering removing them, but I looked through the code base and couldn't find any precedent of test steps without checks, so I left them in.